pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/copyanswer/AgreementAnswerCopier.java

```diff
@@ -0,0 +1,20 @@
+package org.broadinstitute.ddp.db.dao.copyanswer;
+
+import org.broadinstitute.ddp.db.dao.ActivityAnswerCopierSql;
+import org.jdbi.v3.core.Handle;
+
+public class AgreementAnswerCopier extends BaseAnswerCopier {
+
+    @Override
+    public AnswerCopyResult copyAnswer(Handle handle, long sourceInstanceId, String sourceQuestionStableId, long destinationInstanceId,
+                                       String destinationQuestionStableId, long createdAtEpochMillis, long lastUpdatedAtEpochMillis,
+                                       CompositeAnswerCopyConfiguration compositeCopyConfiguration) {
+        AnswerCopyResult baseCopyResult = super.copyAnswer(handle, sourceInstanceId, sourceQuestionStableId, destinationInstanceId,
+                destinationQuestionStableId, createdAtEpochMillis, lastUpdatedAtEpochMillis, compositeCopyConfiguration);
+
+        handle.attach(ActivityAnswerCopierSql.class).copyAgreementAnswer(baseCopyResult.getSourceAnswerId(),
```

Comment:
How about we pass in an instance of `ActivityAnswerCopierSql` into the constructor instead of calling `handle.attach()`? We should already have one when constructing the copiers in the `switch` statement, e.g. `new TextAnswerCopier(this)`. Once you have that, you can also get rid of the `Handle` argument since we can call `copierSql.getHandle()`.

Reply:
Interesting idea!  I'll give this a whirl.  One concern with that approach is that it might be easier for clients to loose track of state and potentially use two different connections when they should only be using one.  Will address this with some javadoc.

Reply:
But if we only pass in `ActivityAnserCopierSql` then the only way to get a connection is to call `copierSql.getHandle()` so I don't see how we might end up with two different connections?